traefik/templates/service.yaml

```diff
@@ -1,45 +1,110 @@
 {{- if .Values.service.enabled -}}
-apiVersion: v1
-kind: Service
-metadata:
-  name: {{ template "traefik.fullname" . }}
-  labels:
-    app.kubernetes.io/name: {{ template "traefik.name" . }}
-    helm.sh/chart: {{ template "traefik.chart" . }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-  annotations:
-  {{- with .Values.service.annotations }}
-  {{- toYaml . | nindent 4 }}
-  {{- end }}
-spec:
-  {{- $type := default "LoadBalancer" .Values.service.type }}
-  type: {{ $type }}
-  {{- with .Values.service.spec }}
-  {{- toYaml . | nindent 2 }}
-  {{- end }}
-  selector:
-    app.kubernetes.io/name: {{ template "traefik.name" . }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-  ports:
-  {{- range $name, $config := .Values.ports }}
-  {{- if $config.expose }}
-  - port: {{ default $config.port $config.exposedPort }}
-    name: {{ $name }}
-    targetPort: {{ $name | quote }}
-    {{- if $config.nodePort }}
-    nodePort: {{ $config.nodePort }}
-    {{- end }}
+
+{{ $tcpPorts := dict }}
+{{ $udpPorts := dict }}
+{{- range $name, $config := .Values.ports }}
+  {{- if eq ($config.protocol | toString) "UDP" }}
+    {{ $_ := set $udpPorts $name $config }}
+  {{- else }}
+    {{ $_ := set $tcpPorts $name $config }}
   {{- end }}
-  {{- end }}
-  {{- if eq $type "LoadBalancer" }}
-  {{- with .Values.service.loadBalancerSourceRanges }}
-  loadBalancerSourceRanges:
-  {{- toYaml . | nindent 2 }}
-  {{- end -}}
-  {{- end -}}
-  {{- with .Values.service.externalIPs }}
-  externalIPs:
-  {{- toYaml . | nindent 2 }}
-  {{- end -}}
+{{- end }}
+
+apiVersion: v1
```

Comment:
Here it seems missing `kind: List`
helm complain : `Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: kind not set`